commit dd59c6676575355dae4da8bd233f428aa582d0d0
Author: James Antill <james@and.org>
Date:   Tue Jul 31 16:35:28 2012 -0400

    Show giant message about mutilib errors, to help people understand. BZ 840543

diff --git a/yum/__init__.py b/yum/__init__.py
index 4650639..6ec5b15 100644
--- a/yum/__init__.py
+++ b/yum/__init__.py
@@ -1184,6 +1184,37 @@ class YumBase(depsolve.Depsolve):
                 if first.verEQ(other):
                     continue
                 msg = _('Protected multilib versions: %s != %s')
+                if not xrestring:
+                    #  People are confused about protected mutilib ... so give
+                    # them a nicer message.
+                    bigmsg = _("""\
+ Multilib version problems found. This often means that the root
+cause is something else and multilib version checking is just
+pointing it that there is a problem. Eg.:
+
+  1. You have an upgrade for %(name)s which is missing some
+     dependency that another package requires. Yum is trying to
+     solve this by installing an older version of %(name)s of the
+     different architecture. If you exclude the bad architecture
+     yum will tell you what the root cause is (which package
+     requires what).
+
+  2. You have multiple architectures of %(name)s installed, but
+     yum can only see an upgrade for one of those arcitectures.
+     If you don't want/need both architectures anymore then you
+     can remove the one with the missing update and everything
+     will work.
+
+  3. You have duplicate versions of %(name)s installed already.
+     You can use "yum check" to get yum show these errors.
+
+...you can also use --setopt=protected_multilib=false to remove
+this checking, however this is almost never the correct thing to
+do as something else is very likely to go wrong (often causing
+much more problems).
+
+""") % {'name' : pkgname}
+                    msg = bigmsg + msg
                 xrestring.append(msg % (first, other))
         if xrestring:
             rescode = 1
commit 3509931c5da4cf62f5080f396b98a70548cf94f4
Author: James Antill <james@and.org>
Date:   Thu Sep 27 14:26:57 2012 -0400

    Fix multilib error typo, and add a TODO for common case #1.

diff --git a/yum/__init__.py b/yum/__init__.py
index a1b045f..4358455 100644
--- a/yum/__init__.py
+++ b/yum/__init__.py
@@ -1186,14 +1186,16 @@ class YumBase(depsolve.Depsolve):
                     bigmsg = _("""\
  Multilib version problems found. This often means that the root
 cause is something else and multilib version checking is just
-pointing it that there is a problem. Eg.:
+pointing out that there is a problem. Eg.:
 
   1. You have an upgrade for %(name)s which is missing some
      dependency that another package requires. Yum is trying to
      solve this by installing an older version of %(name)s of the
      different architecture. If you exclude the bad architecture
      yum will tell you what the root cause is (which package
-     requires what).
+     requires what). You can try redoing the upgrade with
+     --exclude %(name)s.otherarch ... this should give you an error
+     message showing the root cause of the problem.
 
   2. You have multiple architectures of %(name)s installed, but
      yum can only see an upgrade for one of those arcitectures.
